### PR TITLE
Search endpoint

### PR DESCRIPTION
**Pull request by AnvarjonShavqiyev**
- Added a search endpoint in the NestJS API to filter users by name or email.
- Added a search input to the user list UI.
- Fetched and updated search results from the backend.

image
image
image

### CODE REVIEW

File: my-react-app/src/components/navbar/Navbar.scss [hunk 1]
```diff
@@ -18,3 +18,49 @@
...
+}
+
+.search-wrapper{
+    background-color: #fff;
+    width: 50%;
+    height: 42px;
+    border-radius: 18px;
+    padding: 0 10px 0 0px;
+    display: flex;
+    .search-input{
```
Comment (VladKetris) [inline my-react-app/src/components/navbar/Navbar.scss:51]: please add empty line before new selector to improve readablility

File: my-react-app/src/components/navbar/Navbar.tsx [hunk 2]
```diff
@@ -1,19 +1,41 @@
-import { useDispatch } from 'react-redux'
```
Comment (VladKetris) [inline my-react-app/src/components/navbar/Navbar.tsx:1]: where all the semicolons are ?

File: my-react-app/src/components/navbar/Navbar.tsx [hunk 3]
```diff
@@ -1,19 +1,41 @@
...
   return (
     <div className='navbar__wrapper'>
         <Container>
           <div className='navbar'>
-            <h2>User management</h2>
+            <div className='user-info'>
+              <Avatar className='user-avatar'>{user?.name[0].toUpperCase()}</Avatar>   
+              <div className='user-profile'>
+                <h4 className='user-name'>{user?.name}</h4>
+                <Link className='profile-link' to='/profile'>Account</Link>
```
Comment (VladKetris) [inline my-react-app/src/components/navbar/Navbar.tsx:32]: please create separate constant with links

File: my-react-app/src/components/navbar/Navbar.scss (hunk 1 above)
Comment (AnvarjonShavqiyev) [reply my-react-app/src/components/navbar/Navbar.scss:51]: Ok, i try to this next time.
I know that, but i forgot this time.
I will write as you said in future

File: my-react-app/src/components/navbar/Navbar.tsx (hunk 3 above)
Comment (AnvarjonShavqiyev) [reply my-react-app/src/components/navbar/Navbar.tsx:32]: i got it

File: my-react-app/src/components/navbar/Navbar.scss (hunk 1 above)
Comment (VladKetris) [reply my-react-app/src/components/navbar/Navbar.scss:51]: just set up 'eslint' and 'prettier' to do it before commit with 'husky'

**Comment by VladKetris** [comment]
why don't you create constant for links?

File: my-react-app/src/components/navbar/Navbar.tsx (hunk 2 above)
Comment (VladKetris) [reply my-react-app/src/components/navbar/Navbar.tsx:1]: why didn't you added semicolons ?

File: my-react-app/src/components/navbar/Navbar.tsx (hunk 3 above)
Comment (AnvarjonShavqiyev) [reply my-react-app/src/components/navbar/Navbar.tsx]: I am going to add it in next tasks, but now i fixed it, created constants file in router and constant for this link

File: my-react-app/src/components/navbar/Navbar.tsx (hunk 2 above)
Comment (VladKetris) [reply my-react-app/src/components/navbar/Navbar.tsx:1]: man, you should add semicolons everywhere, not only for imports in this file
set up eslint and prettier, it'll do it for you automatically

**Comment by VladKetris** [comment]
ok, please set up eslint and prettier in separate PR

**Review by VladKetris — APPROVED** [review]

**Pull request closed without merging**